pkg/cmd/infra/deployer/deployer.go

```diff
@@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 	} else {
 		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
 	}
-	return strategy.Deploy(from, to, desiredReplicas)
+	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
+		return err
+	}
+	return d.after(config, to)
+}
+
+// AfterDeployment is the default post deployment behavior
+type AfterDeployment struct {
+	// tags allows setting image stream tags
+	tags client.ImageStreamTagsNamespacer
+}
+
+// Handle processes the post-deployment phase
+func (a *AfterDeployment) Handle(config *deployapi.DeploymentConfig, to *kapi.ReplicationController) error {
+	after := config.Spec.Strategy.After
+	if after == nil {
+		return nil
+	}
+
+	var errs []error
+	if len(after.TagImages) > 0 {
+		glog.Infof("Tagging images for successful deployment")
+		for _, action := range after.TagImages {
+			image, err := deployapi.TemplateImageForContainer(&to.Spec.Template.Spec, deployapi.DeploymentConfigHasTrigger(config), action.ContainerName)
```

Comment:
Is the deployment config guaranteed to have the latest image field set?  I
don't know that the annotation of the RC is guaranteed to have that - if
you're confident that the annotated DC on the RC is always correct, then
yes, we should use the value of the image coming from the annotated DC on
the RC.  But it shouldn't come from the DC itself.

On Mon, Feb 1, 2016 at 11:39 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/cmd/infra/deployer/deployer.go
> <https://github.com/openshift/origin/pull/6925#discussion_r51442550>:
>
> > +	// tags allows setting image stream tags
> > +	tags client.ImageStreamTagsNamespacer
> > +}
> > +
> > +// Handle processes the post-deployment phase
> > +func (a *AfterDeployment) Handle(config *deployapi.DeploymentConfig, to *kapi.ReplicationController) error {
> > +	after := config.Spec.Strategy.After
> > +	if after == nil {
> > +		return nil
> > +	}
> > +
> > +	var errs []error
> > +	if len(after.TagImages) > 0 {
> > +		glog.Infof("Tagging images for successful deployment")
> > +		for _, action := range after.TagImages {
> > +			image, err := deployapi.TemplateImageForContainer(&to.Spec.Template.Spec, deployapi.DeploymentConfigHasTrigger(config), action.ContainerName)
>
> Where exactly the deployer is deviating the podspec of the latest rc from
> that of the dc? I would say that having a podspec in a rc different from
> the podspec in the dc is a bigger problem. I also think we are clear on
> direct manipulation of dc-owned rcs being a wrong thing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6925/files#r51442550>.
>